### PR TITLE
adding the remove digit algorithm

### PR DESCRIPTION
**Pull request by meg-1**
### Describe your change:

this algorithm returns the biggest possible result that can be achieved by removing one digit from the given number.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/remove_digit.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    222222
+    >>> remove_digit("2222222")
+    Traceback (most recent call last):
+    TypeError: only integers accepted as input
+    >>> remove_digit("string input")
+    Traceback (most recent call last):
+    TypeError: only integers accepted as input
+    """
+
+    if type(num) == int:
```
Comment (CaedenPH) [inline maths/remove_digit.py:24]:
```suggestion
    if isinstance(num, int):
```

File: maths/remove_digit.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            [char for char in num_str] for char in range(len(num_str))
+        ]
+        for index in range(len(num_str)):
+            num_transpositions[index].pop(index)
+        return sorted(
+            [
+                int("".join([char for char in transposition]))
+                for transposition in num_transpositions
+            ]
+        )[-1]
```
Comment (CaedenPH) [inline maths/remove_digit.py:36]:
```suggestion
        return sorted(
                 int("".join([char for char in transposition]))
                for transposition in num_transpositions,
                reverse=True
            ]
        )[0]
```

**Review by CaedenPH — APPROVED** [review]
@cclauss Looks good - 
Small changes 
  - list comp to generator
  - `type` to `isinstance`

**Comment by meg-1** [comment]
credits to @CaedenPH and @alexpantyukhin for the feedback from #7995 ! just implemented it :)

**Review by ChrisO345 — APPROVED** [review]

**Pull request merged**